editor/src/tools/workers/thumbnail/mesh.ts

```diff
@@ -0,0 +1,106 @@
+import "babylonjs-loaders";
+import { Engine, Scene, FreeCamera, Vector3, CubeTexture, LoadAssetContainerAsync } from "babylonjs";
+
+// TODO: update assimp to work in a worker. Fbx etc. is not the most used format in web, can be delayed.
+// import "../../../loader/assimpjs";
+
+import { readBlobAsDataUrl } from "../../tools";
+
+let engine: Engine;
+let scene: Scene;
+let camera: FreeCamera;
+let environmentTexture: CubeTexture;
+
+export async function getPreview(absolutePath: string, rootUrl: string, serializedEnvironmentTexture?: any) {
+	if (!engine) {
+		engine = new Engine(new OffscreenCanvas(256, 256), true, {
+			antialias: true,
+			audioEngine: false,
+			limitDeviceRatio: 1,
+			adaptToDeviceRatio: false,
+		});
+
+		scene = new Scene(engine);
+		scene.clearColor.set(0, 0, 0, 1);
+		camera = new FreeCamera("camera", new Vector3(100, 300, 100), scene);
+
+		scene.createDefaultLight();
+		scene.createDefaultEnvironment({
+			createSkybox: false,
+			enableGroundShadow: true,
+			enableGroundMirror: true,
+			environmentTexture: scene.environmentTexture!,
+		});
+
+		if (serializedEnvironmentTexture) {
+			environmentTexture = CubeTexture.Parse(serializedEnvironmentTexture, scene, rootUrl);
+			scene.environmentTexture = environmentTexture;
+		}
+	}
+
+	const container = await LoadAssetContainerAsync(absolutePath, scene);
+	container.addAllToScene();
+
+	return new Promise<string>((resolve) => {
+		scene.executeWhenReady(async () => {
+			const center = Vector3.Zero();
+			const minimum = new Vector3(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE);
+			const maximum = new Vector3(Number.MIN_VALUE, Number.MIN_VALUE, Number.MIN_VALUE);
+
+			container.meshes.forEach((mesh) => {
+				mesh.refreshBoundingInfo({
+					applyMorph: true,
+					applySkeleton: true,
+					updatePositionsArray: true,
+				});
+
+				const bb = mesh.getBoundingInfo().boundingBox;
+
+				maximum.x = Math.max(bb.maximumWorld.x, maximum.x);
+				maximum.y = Math.max(bb.maximumWorld.y, maximum.y);
+				maximum.z = Math.max(bb.maximumWorld.z, maximum.z);
+
+				minimum.x = Math.min(bb.minimumWorld.x, minimum.x);
+				minimum.y = Math.min(bb.minimumWorld.y, minimum.y);
+				minimum.z = Math.min(bb.minimumWorld.z, minimum.z);
+
+				center.addInPlace(bb.centerWorld);
+			});
+
+			center.scaleInPlace(1 / container.meshes.length);
+
+			const fov = camera.fov;
+			const aspect = camera.getEngine().getAspectRatio(camera);
+			const sizeVec = maximum.subtract(minimum);
+
+			const verticalSize = sizeVec.y;
+			const horizontalSize = sizeVec.x;
+
+			const verticalDistance = verticalSize / 2 / Math.tan(fov / 2);
+			const horizontalDistance = horizontalSize / 2 / Math.tan((fov * aspect) / 2);
+			const idealDistance = Math.max(verticalDistance, horizontalDistance) * 2;
```

Comment:
That's a good idea

Reply:
I have to factorize, it's the same code used in the preview to focus and travel to an object using CmdOrCtrl+f